sklearn/datasets/_base.py

```diff
@@ -186,6 +198,14 @@ def load_files(container_path, *, description=None, categories=None,
         filenames: ndarray
             The filenames holding the dataset.
     """
+
+    try:
+        assert not (allowed_extensions and ignored_extensions)
+    except AssertionError:
+        raise AssertionError("Ignored extensions and allowed extensions cannot"
+                             " both be present. Please choose one or the"
+                             " other.")
+
```

Comment:
I think we should convert `allowed_extensions` and `ignored_extensions` to a set now, so that `_check_valid_document` is using sets for element inclusion:

```py
allowed_extensions = None if allowed_extensions is None else set(allowed_extensions)
ignored_extensions = None if ignored_extensions is None else set(ignored_extensions)
```